### ISSUE TITLE
LISVAP - RuntimeError: NetCDF: HDF error

### ISSUE THREAD

**Issue by farinfa** [ec-jrc/lisflood-lisvap#29]
Hello,

I am running LISVAP using a series of different climate projections output. The model runs fine with almost all of them except 1 that returns an error that I did not understand:

`File "/eos/jeodpp/home/users/farinfa/conda_lisvap/bin/lisvap", line 14, in <module>
    sys.exit(main())

  File "/eos/jeodpp/home/users/farinfa/conda_lisvap/lib/python3.8/site-packages/lisvap1.py", line 90, in main lisvapexe(lissettings)

  File "/eos/jeodpp/home/users/farinfa/conda_lisvap/lib/python3.8/site-packages/lisvap1.py", line 65, in lisvapexe dynfmw.run()

  File "/eos/jeodpp/home/users/farinfa/conda_lisvap/lib/python3.8/site-packages/pcraster/framework/dynamicFramework.py", line 65, in run self._runDynamic()

  File "/eos/jeodpp/home/users/farinfa/conda_lisvap/lib/python3.8/site-packages/pcraster/framework/frameworkBase.py", line 371, in _runDynamic self._userModel().dynamic()

  File "/eos/jeodpp/home/users/farinfa/conda_lisvap/lib/python3.8/site-packages/lisvap/lisvapdynamic.py", line 78, in dynamic self.readmeteo_module.dynamic()

  File "/eos/jeodpp/home/users/farinfa/conda_lisvap/lib/python3.8/site-packages/lisvap/hydrological/readmeteo.py", line 60, in dynamic self.var.Rds = readnetcdf(self.settings.binding['RdsMaps'], self.var.currentTimeStep())

  File "/eos/jeodpp/home/users/farinfa/conda_lisvap/lib/python3.8/site-packages/lisvap/utils/readers.py", line 198, in readnetcdf current_ncdf_index = netcdf_step(averageyearflag, nf1, timestampflag, timestep)

  File "/eos/jeodpp/home/users/farinfa/conda_lisvap/lib/python3.8/site-packages/lisvap/utils/readers.py", line 228, in netcdf_step t_steps = nf1.variables['time'][:]  # get values for timesteps ([  0.,  24.,  48.,  72.,  96.])

  File "netCDF4/_netCDF4.pyx", line 4408, in netCDF4._netCDF4.Variable.__getitem__

  File "netCDF4/_netCDF4.pyx", line 5352, in netCDF4._netCDF4.Variable._get

  File "netCDF4/_netCDF4.pyx", line 1887, in netCDF4._netCDF4._ensure_nc_success

RuntimeError: NetCDF: HDF error`

I went trough the input files and all seems ok. Can anyone suggest what might be wrong?

Thank you!

**Comment by emiliano-gelati** [comment]
Hi @farinfa,
Does the exception occur on the 1st time step? Also, could you try to re-run LISVAP several times and check if it crashes always on the same time step?

**Comment by farinfa** [comment]
Hi @emiliano-gelati,

yes, always at the first time step. It doesn't even start basically. 
I tried several times and I get always the same error at the same time. 

Thanks!

**Comment by emiliano-gelati** [comment]
Hi @farinfa ,
Are you running LISVAP on the JEODPP terminal service?
To try to reproduce your error, I would need your conda environment definitions (.yml file?) and your settings XML causing the crash

**Comment by farinfa** [comment]
Hi @emiliano-gelati ,

Yes, I'm running LISVAP on JEODPP.
You can find the settings file `settings_FF_remo_ipsl.xml`  in `/eos/jeodpp/data/projects/WEFE/LISFLOOD-Africa/cordex/GCM/FF_settings_lisvap/`.
Regarding the virtual environment definition, I did not manage to find the `.yml` file you requested, but you can find my conda environment in `/eos/jeodpp/home/users/farinfa/conda_lisvap` or you can reproduce the exact installation following [this procedure](https://github.com/ec-jrc/lisflood-lisvap/issues/26#issuecomment-603295771).

Please, let me know if you need anything else.

Thanks!

**Comment by farinfa** [comment]
Hi @emiliano-gelati and @domeniconappo,

I have finally realized that my `rsds.nc` file was corrupted. Something went wrong when I moved the data to the JEODPP.
I couldn't spot the error before, because the file size was exactly the same as the others and the ncdump was giving the expected output. I've realized only interrogating the file with a cdo -ntime/-info. CDO could not open the file and was giving the error: 

`Error (cdf_get_var1_double) : NetCDF: HDF error
Error (cdf_close) : NetCDF: HDF error`

I downloaded the original CORDEX data from ESGF again, re-processed the variable to get the rsds I needed, and now LISVAP is running just fine.
Thanks again for your help.

F

Sorry for not realizing it earlier and for bothering you with this.
